pkg/provisioningv2/kubeconfig/manager.go

```diff
@@ -222,7 +222,7 @@ func (m *Manager) GetCRTBForAdmin(cluster *v1.Cluster, status v1.ClusterStatus)
 func (m *Manager) getKubeConfigData(clusterNamespace, clusterName, secretName, managementClusterName string) (map[string][]byte, error) {
 	secret, err := m.secretCache.Get(clusterNamespace, secretName)
 	if err == nil {
-		if secret.Data == nil || secret.Data["token"] == nil || secret.Annotations["objectset.rio.cattle.io/owner-gvk"] != "provisioning.cattle.io/v1, Kind=Cluster" {
```

Comment:
Checking the annotation seems a little bit problematic during backup and restore as the secret can be created just with no annotation first https://github.com/rancher/rancher/blob/e201d2e02ead6884d789024c7fbacf0819b68363/pkg/provisioningv2/kubeconfig/manager.go#L282. This causes secret to be created and deleted back and forth and somehow the token inside it become invalid. @nickgerace I feel like we can remove this as checking `token` field should suffice.

Reply:
Ahhh good call. That discrepancy makes sense. Let's do it.